solr/core/src/java/org/apache/solr/metrics/prometheus/SolrPrometheusRegistry.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.solr.metrics.prometheus;
+
+import com.codahale.metrics.Meter;
+import com.codahale.metrics.Timer;
+import io.prometheus.metrics.core.metrics.Counter;
+import io.prometheus.metrics.core.metrics.Gauge;
+import io.prometheus.metrics.model.registry.PrometheusRegistry;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+public abstract class SolrPrometheusRegistry {
```

Comment:
What is the lifecycle of these?  In other words, are they created temporarily to serve a metrics request, or are they lived longer?  I'm a little concerned about synchronization if it lives longer (i.e. when a core goes away, etc.)

Reply:
They are just created temporarily to serve metrics requests so keeping anything in sync will not be an issue. Dropwizard holds all the metrics. We just export the registry to Prometheus when called.

Reply:
Changed name from PrometheusRegistry to Exporter to better understand we are just exporting/transforming dropwizard to prometheus instead of keeping separate registries.